### ISSUE TITLE
Be able to assign RPC port

### ISSUE THREAD

**Issue by mccjul** [metaplex-foundation/amman#55]
So I have a particular use case in which I am running amman in a docker container and spawning it to do testing of my contracts and apis via heroku. I am able to make the container work however, heroku does not respect the docker EXPOSE keywords and requires me to inject a port for the app to listen to. solana-test-validator has the --rpc-port command and it seems like it would work for this. I could add this to the project myself via a validator mapping but I would like to have feedback as another way would be to add extra arguments. Ex: amman start --extra="--rpc-port=8080"

**Comment by thlorenz** [comment]
Ah ok .. that requires a change in amman and amman-client.
Basically making this configurable on start and being able to override `LOCALHOST`.

If you want please take a stab at adding this. I'd prefer this as a property on the validator config (vs. a command line arg), but we if you need this then we can add an arg that can override that config value per `amman start`.
That could be as simple as `amman start --rpc-port=8080` or maybe `amman start --rpc-url=http://<some url>:<some-port`. (I don't like the `--extra` way too much).

**Comment by mccjul** [comment]
Yeah thats what I thought. Np! I had expected to do it myself as it is such a specific use-case.

**Comment by staccDOTsol** [comment]
Yes, you can assign an RPC port to the Solana test validator using the `--rpc-port` flag. To specify the port number, simply add `--rpc-port=<port number>` to the command line when starting the validator. For example, to start the validator on port 8080, you would run the following command:

```
solana-test-validator --rpc-port=8080
```

If you want to pass extra arguments to the `solana-test-validator` command, you can use the `--` separator to separate the arguments for the validator from the arguments for the `solana-test-validator` command. For example, to start the validator on port 8080 and pass an extra argument to the validator, you would run the following command:

```
solana-test-validator --rpc-port=8080 -- --extra="--my-extra-argument"
```

I hope this helps! Let me know if you have any further questions.
